### PR TITLE
Raffle Command

### LINKED ISSUE #880: Raffle Command
`s!raffle [amount] [role]` - Raffles X users from the specified role. If no role is specified, raffle within the whole server.

### PR DESCRIPTION
**Pull request by jaxxibae**
+ Closes #880 

image
image
image

### CODE REVIEW

**Comment by dpaiv0** [comment]
> wheres me in the first pic?

You weren't picked by Shieldblade. Dismissing review.

**Comment by dpaiv0** [comment]
Don't approve this yet, I didn't see the first print screen. @pedrofracassi is there twice, and that's outrageous.

**Comment by pedrofracassi** [comment]
Converting this one to a draft.

**Comment by Doges** [comment]
stale?

File: src/commands/misc/raffle.js
Comment (Tsugami) [inline src/commands/misc/raffle.js:28]:
```js
  let raffledUsers = users.filter(m => !m.user.bot).map(u => u.toString())
  const chosen = []

    for (let i = 0; i < amount; i++) {
      const user = raffledUsers[Math.floor(Math.random() * raffledUsers.length)]
      chosen.push(user)
      raffledUsers = raffledUsers.filter(u=> u !== user)
    }
```
solve the problem of duplicate users

**Review by Tsugami — CHANGES_REQUESTED** [review]

**Pull request closed without merging**
